SpatialCrossMapLRN.lua

```diff
@@ -33,9 +33,12 @@ function LRN:createIODescriptors(input)
    end
    assert(input:dim() == 4 and input:isContiguous());
    if not self.iDesc or
-      input:size(1) ~= self.iSize[1] or input:size(2) ~= self.iSize[2]
-   or input:size(3) ~= self.iSize[3] or input:size(4) ~= self.iSize[4] then
+      input:size(1) ~= self.iSize[1] or input:size(2) ~= self.iSize[2] or
+      input:size(3) ~= self.iSize[3] or input:size(4) ~= self.iSize[4] or 
+   not self.iType or input:type() ~= self.iType then
```

Comment:
Is this `not` necessary? If `self.iType == nil` the next condition will evaluate to `true` anyway